network/src/connectivity_manager/mod.rs

```diff
@@ -55,6 +55,8 @@ use tokio::time;
 #[cfg(test)]
 mod test;
 
+const MAX_FULLNODE_CONNECTIONS: usize = 3;
+
```

Comment:
let's make this a more generic parameter, so you can pass in `Option<usize>` to `ConnectivityManager`, where `None` is no limit and `Some(limit)` will limit the number of connections to `limit`.

Reply:
I guess if you do that... then the logic below can change...
```
    let to_connect = min(limit.or_else(to_connect.len()), to_connect.len());
```

Reply:
> I guess if you do that... then the logic below can change...
> 
> ```
>     let to_connect = min(limit.or_else(to_connect.len()), to_connect.len());
> ```

Sadly no, because then that just limits to "3 dials started at one time" rather than 3 connections total.  Have to add in the calculation including inflight connections and dials.

Reply:
this makes more sense to me, I don't like distinguish by role

Reply:
It's still random, and there is still a limit by role, but it's controlled outside of the connectivity manager, and shouldn't be exposed there.